Sandbags Update 2024

Bawhoppen opened the pull request:
Continuation of my old PR https://github.com/tgstation/tgstation/pull/17108

Adds new sprites for sandbags from TGMC, that are slightly modified to fit /tg/ better. They aren't perfect, since they have a bit more of a gritty and rustic feeling compared to our crisp and cartoonish art style, but it's certainly a vast improvement over the previous ones.

Old sprites:
![image](https://github.com/user-attachments/assets/27cf61f1-2adf-45b2-a380-4d9f49c92ba7)

New sprites:
![image](https://github.com/user-attachments/assets/dd3445f9-1f3a-43fb-8c40-20f50b725af5)

These sprites are for the structure themselves, items, and inhands, as seen in the above photo.



Additionally, I made empty sandbags into a stack item, added a few extra to the empty sandbag boxes, and made them occasionally drop leftovers when they're destroyed. 

:cl:
add: Sandbags Update 2024: Sandbags have new sprites, also drop dust + bag & cloth debris now when destroyed, and there are extra empty sandbags in their bag boxes.
/:cl:

Bawhoppen committed:
Icons 1

Bawhoppen committed:
Sanbgs

Bawhoppen committed:
Fix pathchange

Bawhoppen committed:
some other Sandbag stuff

Bawhoppen committed:
sandbag drop tweak

Bawhoppen commented:
Apparently people added sandbags to their maps. Who knew? I'll fix that shortly.

carlarctg commented:
holy shit CM jumpscare

i absolutely hate that this brute forces the railing/windowlike directional sprites into being center tiles. it feels so fucking wrong. either change sandbags to work like that or go home

carlarctg commented:
oh yeah these are from CM not TGMC

Bawhoppen commented:
> holy shit CM jumpscare
> 
> i absolutely hate that this brute forces the railing/windowlike directional sprites into being center tiles. it feels so fucking wrong. either change sandbags to work like that or go home

While I too wish they were directional and not center-tile, I don't think it's kosher with the maintainers nowadays to add more semi-tile structures... So I figured that good is better than perfect, and an improvement for this is worthwhile.

necromanceranne commented:
Why not ask, we have railings and these kind of function like railings but with density.

Bawhoppen commented:
> Why not ask, we have railings and these kind of function like railings but with density.

I suppose I could have asked, but from my understanding, is not that it's impossible, but that it adds far more tech debt than it's worth. But if a maintainer says otherwise, I will add directionals in a future PR.

necromanceranne commented:
We already _have_ railings, what more debt is there to add?

Bawhoppen commented:
How they interact with projectiles is a big one.

LemonInTheDark commented on code/game/machinery/deployable.dm:
```diff
@@ -132,6 +132,14 @@
 	AddElement(/datum/element/elevation, pixel_shift = 12)
 
+/obj/structure/barricade/sandbags/atom_deconstruct(disassembled = TRUE)
+	var/atom/Tsec = drop_location()
```
this name is real bad. what is it exactly? is it a contraction if so just put the whole thing

Bawhoppen replied:
It seemed like the standard that other many other structures used for this, so I went with it. Anything better that I should change it to?

LemonInTheDark replied:
well like, what is the var? is it the spawn location? is it the place we're dropping stuff?

Bawhoppen replied:
In truth you may not like this answer: this is how it's done in so many other places in the code, I figured I would just follow along and do the same. If I should do something else I will.

LemonInTheDark replied:
please give it a name that describes what it actually does here

LemonInTheDark commented on code/game/objects/items/stacks/sheets/mineral.dm:
```diff
@@ -69,24 +73,38 @@ GLOBAL_LIST_INIT(sandbag_recipes, list ( \
 
-/obj/item/emptysandbag
-	name = "empty sandbag"
+/obj/item/stack/sheet/mineral/emptysandbags
```
why make empty sandbags a mineral?

Bawhoppen replied:
There really is no reason other than to keep them similar to the filled sandbags which are also under the mineral path... I am sure there was some reason I did that originally but I don't remember why.

LemonInTheDark commented on code/game/objects/items/stacks/sheets/mineral.dm:
```diff
@@ -69,24 +73,38 @@ GLOBAL_LIST_INIT(sandbag_recipes, list ( \
+
+/obj/item/stack/sheet/mineral/emptysandbags/attackby(obj/item/W, mob/user, params)
+	add_fingerprint(user)
 	if(istype(W, /obj/item/stack/ore/glass))
```
early return please (check the inverse of this and call return ..() to avoid indentation)

LemonInTheDark commented on code/game/objects/items/stacks/sheets/mineral.dm:
```diff
@@ -69,24 +73,38 @@ GLOBAL_LIST_INIT(sandbag_recipes, list ( \
-		G.use(1)
-	else
-		return ..()
+		if (G.get_amount() >= 1 && get_amount() >= 1)
```
similar

LemonInTheDark commented on code/game/objects/items/stacks/sheets/mineral.dm:
```diff
@@ -69,24 +73,38 @@ GLOBAL_LIST_INIT(sandbag_recipes, list ( \
+/obj/item/stack/sheet/mineral/emptysandbags/attackby(obj/item/W, mob/user, params)
+	add_fingerprint(user)
 	if(istype(W, /obj/item/stack/ore/glass))
 		var/obj/item/stack/ore/glass/G = W
```
please rename this to use a non single letter var name, while we're here

LemonInTheDark commented on code/game/objects/items/stacks/sheets/mineral.dm:
```diff
@@ -69,24 +73,38 @@ GLOBAL_LIST_INIT(sandbag_recipes, list ( \
-		return ..()
+		if (G.get_amount() >= 1 && get_amount() >= 1)
+			to_chat(user, span_notice("You fill the sandbag."))
+			var/obj/item/stack/sheet/mineral/sandbags/I = new (get_turf(user))
```
similar

LemonInTheDark commented on code/game/objects/items/stacks/sheets/mineral.dm:
```diff
@@ -69,24 +73,38 @@ GLOBAL_LIST_INIT(sandbag_recipes, list ( \
+			var/obj/item/stack/sheet/mineral/sandbags/I = new (get_turf(user))
+			if(!QDELETED(I))
+				I.add_fingerprint(user)
+			var/replace = user.get_inactive_held_item() == src
```
why is this a var? can't we just put it in the code?

LemonInTheDark commented on code/game/objects/items/storage/boxes/security_boxes.dm:
```diff
@@ -225,8 +225,8 @@
 /obj/item/storage/box/emptysandbags/PopulateContents()
-	for(var/i in 1 to 7)
-		new /obj/item/emptysandbag(src)
+	for(var/i in 1 to 2)
+		new /obj/item/stack/sheet/mineral/emptysandbags/fifty(src)
```
wayyyy more, intentionally? why?

Bawhoppen replied:
The only reason for that is because #1, I wanted to have more than one stack in a box or else it would look kind of awkward, but I still wanted the max amount per stack to be 50, #2 I don't think it matters at all since I have never seen them used literally ever since I added them

LemonInTheDark commented:
![image](https://github.com/user-attachments/assets/3edbdd5f-e96c-4f35-9344-e777714f7817)
this layering kinda blows, should sandbags be on a higher layer?

necromanceranne commented:
> ![image](https://private-user-images.githubusercontent.com/58055496/371925349-3edbdd5f-e96c-4f35-9344-e777714f7817.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc2NjQxMzEsIm5iZiI6MTcyNzY2MzgzMSwicGF0aCI6Ii81ODA1NTQ5Ni8zNzE5MjUzNDktM2VkYmRkNWYtZTk2Yy00ZjM1LTkzNDQtZTc3NzcxNGY3ODE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTMwVDAyMzcxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZhYzE0YzQ0YjgwYTg1NmQ2ZGQ5ZjhkMzVkYmY3N2VhOGVhNGRhMTE3ZTFmNDY1YjkyZmZlNjFlNjg1ODhlZjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.hWsIPJgAqLuZU9LDtquwT0qAWRW09_dkK3UJgvFvh9Q) this layering kinda blows, should sandbags be on a higher layer?

I think that's because they're standing on it.

Bawhoppen commented:
They indeed are standing on it in that example. There is no overlap for the sprite, it's fully contained within the tile.

iwishforducks commented:
These look much better on their own but I really think these need to be directional. It's not immediately obvious to me that they block the entire tile.

Also these don't look like proper sandbag emplacements since they're not properly "brick" stacked. They just kinda look like they would topple over if somebody walked into them. Don't think that could be fixed without completely respriting them, though.

Likteer commented:
This feels like a downgrade to me.

Loafin34 commented:
nah... the sandbags are good the way they are because they can't be directional like that without looking like ass.

SimplySmiley123 commented:
These sprites would need to be directional to make sense.
Too thin to block the whole tile.

Bawhoppen committed:
Fixes some sb code

Wallemations reviewed (changes requested):
A couple things

1. These are far too thin, where when people stand on top of them it just looks like they're clipping into them.
2. They really should have smoothing applied to them, since as it stands they look super weird when they're vertically stacked near each other.
3. These feel like they have way too much contrast in their palettes. They really stand out, and not in a good way. Since this is a structure, you should look at the wallening to see what palettes are being used there. Maybe sandstone structures could help for that?

Bawhoppen commented:
> A couple things
> 
> 1. These are far too thin, where when people stand on top of them it just looks like they're clipping into them.
> 2. They really should have smoothing applied to them, since as it stands they look super weird when they're vertically stacked near each other.
> 3. These feel like they have way too much contrast in their palettes. They really stand out, and not in a good way. Since this is a structure, you should look at the wallening to see what palettes are being used there. Maybe sandstone structures could help for that?

1. There is no way to make them less thin without a full resprite of which I am incapable. I can however adjust the standing offset a bit to make it a little better.

2. There's a few aspects to this: For one, to be candid, I don't have the spriting skills to make them fully smooth- I tried to my best to tweak them to still look good while working with the original base and making them semi-smooth. Beyond that, in most uses of these there will only be a couple placements next to each other, so it will be less severe than the example image. But even then, I think the fact they are separated a bit actually works. While real sandbags are piled up in straight lateral lines, they also are often set up in small segments adjacent each other (though those often have posts separating them). The fact that if we imagine individual sand bags as part of the structure, also helps make it appear as less weird. So while probably not perfect, it doesn't seem that bad to me. 
 
3. I already toned down their contrast from the original TGMC/CM sprites, but I'll see if I can do more without compromising their quality. That being said, I personally don't think they stand out much at all when ingame. I think the reference images draw more attention to them than in normal day-to-day use.

Wallemations commented:
> There is no way to make them less thin without a full resprite of which I am incapable. I can however adjust the standing offset a bit to make it a little better.

If you can't resprite it or find someone else to handle addressing the issues that make it so out of place, I'll be forced to close the PR.

Bawhoppen commented:
> > There is no way to make them less thin without a full resprite of which I am incapable. I can however adjust the standing offset a bit to make it a little better.
> 
> If you can't resprite it or find someone else to handle addressing the issues that make it so out of place, I'll be forced to close the PR.
> 
> I'll give you a few days to figure that out, I'm sure someone in the sprite channel could help you out, though I can't be 100% sure they'll be free to do the resprite themselves.

So this is not a significant enough improvement on its own to warrant merging?

nikothedude commented:
their contrast really is insane yeah

Bawhoppen commented:
I spoke to Wallemations and the sprite will be worked on to address the worst issues.

zxaber commented:
What about just removing standing on top of them? Dragging your sprite onto them could have you vault over to the other side instead.

Bawhoppen commented:
Regrettably, after some time approaching it, I don't think there's any way I'm going to be able to modify the sprites myself to where they look good to the sprite maintainers. I can still adjust the standing offset which I think would fix it, but I don't think that's an option.
If someone comes up with some sprites or some solution (Zxaber your idea was good but won't work for a few reasons), send them to me and I'll reopen this PR and fix the minor path issues so it can be merged.
